### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by profxj**
Major refactor of the PypeIt naming.

Also integrates (not fully) the new `SpecObj` and
`SpecObjs` objects.

### CODE REVIEW

**Comment by profxj** [comment]
Has passed at least one Dev-suite tests on the new 
`rename_me` branch there.

**Comment by profxj** [comment]
passing tests on laptop;  still working to turn Travis on

**Comment by jhennawi** [comment]
I think these conflicts have to do with the fact that we were working in the pypit/core directory which became pypeit/core. Possibly a mistake on our end.

**Comment by profxj** [comment]
All Dev suite passing except WHT (bombs in flexure)

**Comment by kbwestfall** [comment]
PS - Tests pass for me, but someone needs to fix the conflicts.

**Comment by rcooke-ast** [comment]
I don't want to bring this up again, but given the name woes discussed today in the telecon, is it painless enough to make a new branch, do a "find-replace all" to change `PypeIt` to say `Pype`, then check nothing breaks?

Honestly, I don't care too much, but it's probably going to be misspoken and confused quite a bit.

File: pypeit/arcimage.py [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
 
-from pypit import msgs
-from pypit import processimages
-from pypit import masterframe
-from pypit.core import arsort
-from pypit.par import pypitpar
+from pypeit import msgs
+from pypeit import processimages
+from pypeit import masterframe
+from pypeit.core import sort
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/arcimage.py]: sort -> fsort

File: pypeit/arms.py [hunk 2]
```diff
@@ -7,29 +7,25 @@
...
-
-from pypit import ardebug as debugger
-
-try:
-    basestring
-except NameError:
-    basestring = str
+from pypeit import msgs
+from pypeit.core import load
+from pypeit import utils
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/arms.py]: beyond my endurance for now.

File: pypeit/arms.py [hunk 3]
```diff
@@ -7,29 +7,25 @@
...
+from pypeit.core import save
+from pypeit.core import wave
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/arms.py]: agreed, it needs a new name.

File: pypeit/pypeit.py [hunk 4]
```diff
@@ -5,31 +5,31 @@
...
-from pypit import pypitsetup
+from pypeit import pypeitsetup
 
-from pypit import ardebug
+from pypeit import debugger
 
-def PYPIT(pypit_file, setup_only=False, calibration_check=False, use_header_frametype=False,
-          sort_dir=None, debug=None, quick=False, ncpus=1, overwrite=False, verbosity=2,
+def PypeIt(pypeit_file, setup_only=False, calibration_check=False, use_header_frametype=False,
+          sort_dir=None, debug=None, quick=False, ncpus=1, overwrite=True, verbosity=2,
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/pypeit.py]: for DRPs, I think overwrite=True is a better default.

**Comment by profxj** [comment]
I fear there was something in arparse.py that I missed
from the X-Shooter bit, but I have it close enough.

**Comment by profxj** [comment]
merging.

**Pull request merged**